packages/aws-cdk/lib/assets.ts

```diff
@@ -47,9 +50,33 @@ async function prepareAsset(asset: AssetMetadataEntry, toolkitInfo: ToolkitInfo,
   }
 }
 
+async function prepareBuildAsset(asset: BuildMetadataEntry, toolkitInfo: ToolkitInfo): Promise<CloudFormation.Parameter[]> {
+  debug('Building project asset:', asset.codePath);
+
+  // working directory within container
+  const wd = '/tmp/cdk.out/';
+
+  await shell([
+    'docker', 'run',
+    ...(asset.stdin !== undefined ? ['-i'] : []),
+    '-w', wd,
+    '--mount', `type=bind,source=${asset.codePath},target=${wd}`,
```

Comment:
Mounts default to be writable back to the host system. Is that what you want to do?